plonky2x/src/frontend/eth/mpt/mpt.rs

```diff
@@ -0,0 +1,310 @@
+use std::marker::PhantomData;
+
+use curta::math::field::Field;
+use itertools::Itertools;
+
+use super::generators::*;
+use crate::prelude::{
+    ArrayVariable, BoolVariable, ByteVariable, Bytes32Variable, CircuitBuilder, PlonkParameters,
+    Variable,
+};
+
+pub fn transform_proof_to_padded<const ENCODING_LEN: usize, const PROOF_LEN: usize>(
```

Comment:
might make more sense to have a struct called EvmStorageProof and impl pad()

Reply:
I do see the argument for this, but going to keep it for now since it's just 1 method on the `Vec<Vec<>>`